.github/workflows/publish-mcp-server.yml

```diff
@@ -129,17 +129,30 @@ jobs:
 
       - uses: actions/setup-node@v4
         with:
-          node-version: 'lts/*'
+          node-version: "lts/*"
           cache: yarn
 
       - name: Install dependencies
         run: yarn install --frozen-lockfile
 
+        # NOTE: This step must be before the "Build package" step
+      - name: Replace AppInsights Key
+        run: |
+          # Replace the empty APP_INSIGHTS_KEY with the actual key from secrets
+          sed -i "s|const APP_INSIGHTS_KEY = '';|const APP_INSIGHTS_KEY = '$APP_INSIGHTS_KEY';|" packages/mcp/src/telemetry.ts
+
+          if grep -q "applicationinsights.azure.com" packages/mcp/src/telemetry.ts; then
```

Comment:
can we make the regex check a bit more strict (a comment with this URL in the file would make it pass), something like capturing `const APP_INSIGHTS_KEY = '<content>'`, then matching the string against the content inside single quotes?

Reply:
Good call 👍

Reply:
Improved the grep. In the screenshot, I added the correct KEY (passed) and then delete a few chars (failed)
<img width="2382" height="846" alt="Screenshot 2025-09-30 at 3 22 20 PM" src="https://github.com/user-attachments/assets/da4c5af4-85db-4e4e-84c6-4fd3d1bfdedd" />